python/lsst/ts/m2/csc.py

```diff
@@ -370,95 +430,186 @@ async def do_standby(self, data: salobj.BaseMsgType) -> None:
+        await self._execute_command(
+            self.controller_cell.set_closed_loop_control_mode,
+            ClosedLoopControlMode.Idle,
+        )
+        await self._execute_command(
+            self.controller_cell.reset_force_offsets,
+        )
+        await self._execute_command(
+            self.controller_cell.reset_actuator_steps,
+        )
 
-    async def begin_disable(self, data: salobj.BaseMsgType) -> None:
-        # multiply timeout by 3 as this is the number of commands executed with
-        # this timeout.
-        await self.cmd_disable.ack_in_progress(data, timeout=self.COMMAND_TIMEOUT * 3)
+        # Power on the system and enable the ILCs
+        await self._execute_command(
+            self.controller_cell.power,
+            PowerType.Communication,
+            True,
+            timeout=self.COMMAND_TIMEOUT_LONG,
+        )
+        await self._execute_command(
+            self.controller_cell.power,
+            PowerType.Motor,
+            True,
+            timeout=self.COMMAND_TIMEOUT_LONG,
+        )
 
-        await super().begin_disable(data)
+        if not self.controller_cell.are_ilc_modes_enabled():
+            try:
+                await self._execute_command(
+                    self.controller_cell.set_ilc_to_enabled,
+                    timeout=self.ilc_timeout,
+                    retry_times=self.ilc_retry_times,  # type: ignore[arg-type]
+                )
 
-    async def do_disable(self, data: salobj.BaseMsgType) -> None:
-        timeout = self.COMMAND_TIMEOUT
+            except RuntimeError:
+                # TODO: Publish the list of failed ILCs after the DM-40146 is
+                # done.
+                await self._basic_cleanup_and_power_off_motor()
+                raise
 
-        await self._transition_controller_state(
-            salobj.State.ENABLED, "disable", timeout
+        await self._execute_command(
+            self.controller_cell.set_closed_loop_control_mode,
+            ClosedLoopControlMode.OpenLoop,
+            timeout=self.COMMAND_TIMEOUT_LONG,
         )
 
-        await self._transition_controller_state(
-            salobj.State.DISABLED, "standby", timeout
+        # Wait for some time before transitioning to the closed-loop control
+        await asyncio.sleep(self.SLEEP_TIME_SHORT)
+
+        await self._switch_force_balance_system(True)
+
+        # Wait for some time to stabilize the system
+        await asyncio.sleep(self.SLEEP_TIME_SHORT)
+
+        # System is ready now. If there is any error, the system will
+        # transition to the Fault state from now on.
+        self.system_is_ready = True
+
+        await super().do_enable(data)
+
+    async def _bypass_monitor_ilc_read_error(self) -> None:
+        """Bypass the error codes related to the monitoring inner-loop
+        controller (ILC) read error before the fix.
+
+        TODO: Remove this after the ILC is fixed.
+        """
+
+        codes = {1000, 1001, 6052}
```

Comment:
What are these "codes"? Are these the codes you are bypassing? if so, consider renaming this `ilc_codes_to_bypass` or something more explanatory.

Reply:
I renamed the variable in commit: [d957e11](https://github.com/lsst-ts/ts_m2/pull/48/commits/d957e11f8ce9f6c4abdfa831a4d402b018c8d11d).

I will remove this function after upgrading the ILC firmware (hope the new version fixed this already).

Thanks!